examples/platform/silabs/wfx_rsi.h

```diff
@@ -106,7 +106,10 @@ int32_t wfx_rsi_disconnect();
 int32_t wfx_wifi_rsi_init(void);
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
 #if SLI_SI91X_MCU_INTERFACE
-void sl_si91x_invoke_btn_press_event();
+void sl_si91x_btn_event_handler();
+void sl_si91x_uart_power_requirement_handler();
+// this is callback from the Wiseconnect SDK
+void gpio_uulp_pin_interrupt_callback(uint32_t pin_intr);
```

Comment:
These functions seem to be specific to the 917 but are in a header file that is common to all wi-fi platforms.
Can we start moving these function out of the "global" header?

Reply:
@mkardous-silabs , do you suggest keeping something like sl_si91x_hal.h instead? Since we do not have anything like that now on this.

Reply:
Call it SiWxPlatformInterface.h for now and we can see later its finale name. 
Put these functions inside a `SiWxPlatform` namespace.